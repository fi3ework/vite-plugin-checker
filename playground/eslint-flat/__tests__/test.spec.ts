import stringify from 'fast-json-stable-stringify'
import { describe, expect, it } from 'vitest'
import {
  diagnostics,
  editFile,
  expectStderrContains,
  isBuild,
  isServe,
  log,
  resetReceivedLog,
  sleepForEdit,
  sleepForServerReady,
  stripedLog,
} from '../../testUtils'

describe('eslint', () => {
  describe.runIf(isServe)('serve', () => {
    it('get initial error and subsequent error', async () => {
      await sleepForServerReady()
      expect(stringify(diagnostics)).toMatchSnapshot()
      expect(stripedLog).toMatchSnapshot()

      console.log('-- edit error file --')
      resetReceivedLog()
      editFile('src/index.js', (code) => code.replace(`Hello`, `Hello~`))
      await sleepForEdit()
      expect(stringify(diagnostics)).toMatchSnapshot()
      expect(stripedLog).toMatchSnapshot()
    })
  })

  describe.runIf(isBuild)('build', () => {
    const expectedMsg = 'Missing semicolon'

    it('should fail', async () => {
      expectStderrContains(log, expectedMsg)
    })
  })
})
