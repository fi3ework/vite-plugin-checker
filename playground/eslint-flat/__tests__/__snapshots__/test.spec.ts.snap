// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`eslint > serve > get initial error and subsequent error 1`] = `"[{"checkerId":"ESLint","frame":"  > 1 | const a = /"hello/"/n      |                  ^/n  > 2 |/n      | ^","id":"<PROJECT_ROOT>/playground-temp/eslint-flat/src/index.js","level":1,"loc":{"column":18,"file":"<PROJECT_ROOT>/playground-temp/eslint-flat/src/index.js","line":1},"message":"Missing semicolon. (semi)","stack":""}]"`;

exports[`eslint > serve > get initial error and subsequent error 2`] = `
" ERROR(ESLint)  Missing semicolon. (semi)
 FILE  <PROJECT_ROOT>/playground-temp/eslint-flat/src/index.js:1:18

  > 1 | const a = "hello"
      |                  ^
  > 2 |
      | ^
[ESLint] Found 1 error and 0 warning"
`;

exports[`eslint > serve > get initial error and subsequent error 3`] = `"[{"checkerId":"ESLint","frame":"  > 1 | const a = /"hello/"/n      |                  ^/n  > 2 |/n      | ^","id":"<PROJECT_ROOT>/playground-temp/eslint-flat/src/index.js","level":1,"loc":{"column":18,"file":"<PROJECT_ROOT>/playground-temp/eslint-flat/src/index.js","line":1},"message":"Missing semicolon. (semi)","stack":""},{"checkerId":"ESLint","frame":"  > 1 | const a = /"hello/"/n      |                  ^/n  > 2 |/n      | ^","id":"<PROJECT_ROOT>/playground-temp/eslint-flat/src/index.js","level":1,"loc":{"column":18,"file":"<PROJECT_ROOT>/playground-temp/eslint-flat/src/index.js","line":1},"message":"Missing semicolon. (semi)","stack":""}]"`;

exports[`eslint > serve > get initial error and subsequent error 4`] = `
" ERROR(ESLint)  Missing semicolon. (semi)
 FILE  <PROJECT_ROOT>/playground-temp/eslint-flat/src/index.js:1:18

  > 1 | const a = "hello"
      |                  ^
  > 2 |
      | ^
[ESLint] Found 1 error and 0 warning"
`;
